### ISSUE TITLE
WebRTC is not working on HTML5 export of Godot 4.2 Dev 

### ISSUE THREAD

**Issue by ghghorbanzade** [godotengine/godot#80056]
### Godot version

4.2.dev

### System information

Windows 10 - NVIDIA GeForce RTX 2080 SUPER [Version 531.29] - Godot 4.2.dev

### Issue description

WebRTC is not working on HTML5 export of Godot 4.2 dev. In testing [the Nakama-WebRTC demo project](https://gitlab.com/snopek-games/godot-nakama-webrtc/-/tree/godot-4?ref_type=heads), I face a problem with Web export. Users get stuck in the connecting state. 

### Steps to reproduce

1- export the project in [this link](https://gitlab.com/snopek-games/godot-nakama-webrtc/-/tree/godot-4?ref_type=heads) for HTML5. ( Test can be done by running the exported project locally or on a web host like itch.io.
2- configure a Nakama server with an SSL certificate and use port forwarding to secure the Nakama port
3- try to authenticate on the exported app.
4- create a room with one client and joint with another one. 
5- It stucks in the connecting state.

### Minimal reproduction project

NA (I didn't manipulate the original project except for port and server address)

**Comment by clayjohn** [comment]
Is this the same as https://github.com/godotengine/godot/issues/80022?

**Comment by ghghorbanzade** [comment]
> Is this the same as #80022?

No, the previous one was related to HttpRequest this one is about making peer-to-peer connections using WebRTC.
The problem rises after authentication is done by HttpRequest.
The first user connects without a problem. But the other users who join the room are stuck in the connecting state. 
I guess something goes wrong when add_peer wants to add the other users' players to the room.

**Comment by clayjohn** [comment]
CC @dsnopek

**Comment by dsnopek** [comment]
I need some more time to investigate, but I strongly suspect this is a bug in my addon rather than Godot itself. On my Discord, another user has been having some trouble with the Godot 4 version of the addon - I was able to fix one of their bugs, but some still remain. I'll try to find some time this week to dig deeper into it.

**Comment by ghghorbanzade** [comment]
> I need some more time to investigate, but I strongly suspect this is a bug in my addon rather than Godot itself. On my Discord, another user has been having some trouble with the Godot 4 version of the addon - I was able to fix one of their bugs, but some remain. I'll try to find some time this week to dig deeper into it.

Actually I made some more tests and it works properly in Chrome browser. But in Mozilla, joining a room stucks in the connecting state. I'm trying to understand what is wrong with Mozilla. 
Thanks for your awesome project anyway. I'm looking forward new modifications of the project.

**Comment by ChrisTrenkamp** [comment]
[webrtc-godot.zip](https://github.com/godotengine/godot/files/13326146/webrtc-godot.zip)

I'm encountering a strange issue with WebRTC as well.  There are two projects in the attached project.

One is a Godot project that uses WebRTC to send packets between peers.  One WebRTC client acts as the dedicated server, and the others that connect to the server are clients.  The clients send packets only to the dedicated server, and the server broadcasts that packet to the other clients.

The other project is a dedicated server that tracks the chatrooms and relays the WebRTC ICE packets for clients that want to connect to a chatroom.  It is written in Go.

The native clients work fine, but the web export partially works.  The web client is able to act as a host and can talk to its connected peers just fine.  However, the web client cannot connect to a hosted server for some reason, and I cannot figure out why.

Here's a screenshot of a working example using only native clients:

![image](https://github.com/godotengine/godot/assets/3903329/162afdde-4336-43df-bf0b-8ad930128276)

Here's a screenshot of the web export acting as the host, and the native client  connected to it:

![image](https://github.com/godotengine/godot/assets/3903329/5a4b09a9-5b67-403e-bd6b-f7f0b2165895)

However, when I set a native client as a host and try to connect to it in the browser, it indefinitely hangs (notice the client doesn't display a "connected" status message in the logs):

![image](https://github.com/godotengine/godot/assets/3903329/3a7440b9-c920-403b-9abd-e3a15442c429)

To reproduce this example:
* [Download and install the Go compiler](https://go.dev/dl/)
* [Download the WebRTC GDExtension](https://github.com/godotengine/webrtc-native/releases/tag/1.0.3-stable)
* Extract the `webrtc` folder into the root of the project.
* Start the dedicated server by running `go build ; ./webrtc-test-server` in the `server` folder.
* Open the Godot project and run the HTML5 export to the `server/static` directory.
* Open a browser to `http://localhost:8080` (the websocket will fail to upgrade if you use `127.0.0.1`),
* Run a native client.
* In the browser client, enter a server name and click `Host`.  In the native client, enter the same server name and click `Connect`.  It should succeed and the two clients can talk to each other.
* Now refresh the browser and restart the native client, but this time, host in the native client and connect in the browser.  The WebRTC ICE exchange does not complete and the browser and native client cannot talk to each other.

**Comment by Faless** [comment]
I can't seem to be able to reproduce using the [webrtc_signaling demo](https://github.com/godotengine/godot-demo-projects/tree/master/networking/webrtc_signaling), tested with both Chromium and Firefox
